### ISSUE TITLE
Refactor XGBQuantileForecaster

### ISSUE THREAD

**Issue by MoritzM00** [MoritzM00/proba-forecasting#12]
Refactor the XGBQuantileForecaster by implementing a RegressionForecaster base class, and XGBQuantileForecaster inherits from this class.
Allows the use of other models more easily, e.g. QuantileRegression from sklearn or other GBT-libraries

Needs to support validation set
